packages/cache/src/index.js

```diff
@@ -68,14 +76,17 @@ let createCache = (options?: Options): EmotionCache => {
   if (isBrowser) {
     container = options.container || document.head
 
-    const nodes = document.querySelectorAll(`style[data-emotion-${key}]`)
+    Array.prototype.forEach.call(ssrStyles, (node: HTMLStyleElement) => {
+      const attrib = node.getAttribute(`data-emotion`).split(' ')
+
+      if (attrib[0] !== key) {
+        return
+      }
+
+      for (let i = 1; i < attrib.length; i++) {
+        inserted[attrib[i]] = true
+      }
 
-    Array.prototype.forEach.call(nodes, (node: HTMLStyleElement) => {
-      const attrib = node.getAttribute(`data-emotion-${key}`)
-      // $FlowFixMe
-      attrib.split(' ').forEach(id => {
-        inserted[id] = true
-      })
       if (node.parentNode !== container) {
```

Comment:
we might need to reparent here what already got moved previously to head if the container is a custom one